migration/migrator/migrations/course/20240205154116_add_poll_timer.py

```diff
@@ -0,0 +1,49 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("ALTER TABLE polls ADD COLUMN IF NOT EXISTS duration INTEGER DEFAULT 0")
+    database.execute("ALTER TABLE polls ADD COLUMN IF NOT EXISTS end_time timestamptz DEFAULT NULL")
+    database.execute("ALTER TABLE polls ALTER COLUMN status DROP NOT NULL")
+    database.execute("ALTER TABLE polls ADD COLUMN IF NOT EXISTS is_visible BOOLEAN NOT NULL DEFAULT FALSE")
+    database.execute("""
+        UPDATE polls SET 
+            end_time = CASE
+                WHEN status = 'open' THEN NULL
+                ELSE now() 
+            END,
+            is_visible = CASE
+                WHEN status IN ('open', 'ended') THEN TRUE
+                ELSE FALSE
+            END
```

Comment:
This can be broken into two separate update statements, and the `end_time` changes can be simplified to something like this:

```
UPDATE polls
SET end_time = now()
WHERE status != 'open'
```

Would it also make sense to set the end time to a date in the past instead of `now()`?

Reply:
if it is closed, the is_visible would be false so the end_time wouldn't matter. If the poll was set to open, the end_time should be null and is_Visible should be true therefore signaling that it is open. If the poll was ended, is_visible would be true but there would be a endtime that is less than (now) so we could change it to beginning of time if that is more suitable

Reply:
Actually, it does make sense for it to be the beginning of time. I will change it to 1900-02-01